### DISCUSSION TITLE
Is float type secure?

### DISCUSSION THREAD

**Discussion by tomasweigenast** [isar/isar#661] [Help]
Hi. I want to know if using custom `float` type is secure and will not loose precision.
Because using ByteData to serialize float32, I have issues when decoding:
```dart
byteData.setFloat32(_offset, 21.25156);
v = byteData.getFloat32(++_offset);
```

Being `v` equals to `21.25156021118164`. You see the difference? 
Event the API alerts you:
![image](https://user-images.githubusercontent.com/31850785/189504555-d2bf1740-93af-4172-a17d-387d895cf3d0.png)

**Comment by simc** [comment]
`Float` has less precision than `double`. It also uses half the space (4 instead of 8 bytes).

**Reply by tomasweigenast** [reply]
Yes, I know, but I wanna know if I will suffer the same issue described above using isar?

**Reply by simc** [reply]
Yes and it is not an issue but the nature of a less precise type. Use double if you don't want that.
